### ISSUE TITLE
Include preview version/changes in filter

### ISSUE THREAD

**Issue by josesimoes** [NuKeeperDotNet/NuKeeper#457]
Hi,

How can we include preview versions in the filter?
I have this situation:
```
Found 1 possible updates
nanoFramework.CoreLibrary from 1.0.1-preview094 to 1.0.1-preview103 in packages.config
```

But I can't persuade the update to happen, it's always being filtered out...

**Comment by josesimoes** [comment]
Looking at the code it seems that the only accepted updates are Major, Minor and Patch...
If I'm understanding this correctly a package with "preview", "alpha" or similar version won't ever be considered as a valid upgrade option.

May I suggest that a new value _Any_ be added to the _change_ option allowing "any" available update package to be considered as a valid update?

**Comment by AnthonySteele** [comment]
That's interesting - we allow betas to be updated to later betas - as above, but we hadn't considered that this would fall below the minimum allowed change!

**Comment by josesimoes** [comment]
I've implemented locally my proposed fix by adding the  Revision value. If you want I can submit a PR with that.

**Comment by AnthonySteele** [comment]
If it works, submit it!

**Comment by skolima** [comment]
Hm, there's a slight naming mismatch here, I think? The prerelease suffix https://docs.microsoft.com/en-us/nuget/reference/package-versioning doesn't seem to have a consistent name to itself across all the documents, but generally, this seems to be called "pre-release".

Internally, NuGet calls them "release labels" or "prerelease labels" in the same code line https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Versioning/NuGetVersion.cs#L88

**Comment by josesimoes** [comment]
@skolima you are correct about that. The naming is pretty... open... Having said that the naming on the "property" in Nuget classes is "Release" that's what I've used in the PR fixing this.

https://github.com/NuGet/NuGet.Client/blob/216a5b237be2f3f923b711c34a39281cdeb1ede7/src/NuGet.Core/NuGet.Versioning/NuGetVersion.cs#L63

(I've amended my comments above from Revision to Release, for consistency sake)

**Comment by AnthonySteele** [comment]
Hm, if the current version is `1.0.1-preview094` then a later preview should be applied, without changes to the enum?The issue might be elsewhere.

**Comment by AnthonySteele** [comment]
For `nanoFramework.CoreLibrary` the last update that I see is `1.0.0-preview073`, 21 days ago: https://www.nuget.org/packages/nanoFramework.CoreLibrary/1.0.0-preview073

Are you using an internal feed with newer versions? Are they under 7 days old? That would filter them out by default. You can disable the package min age check with `--age 0`
